packages/koa-performance/src/test/middleware.test.ts

```diff
@@ -0,0 +1,546 @@
+import {createMockContext} from '@shopify/jest-koa-mocks';
+import {StatusCode, Method, Header} from '@shopify/network';
+import {
+  LifecycleEvent,
+  EventType,
+  Navigation,
+  NavigationDefinition,
+  NavigationResult,
+  NavigationMetadata,
+} from '@shopify/performance';
+import withEnv from '@shopify/with-env';
+
+import {clientPerformanceMetrics, Metrics, TagValue} from '../middleware';
```

Comment:
This file was almost entirely copied from web directly.

It might actually be a nice candidate to also have some more integration style `rack.mount` style testing like we did in the react-server tests, that way we could test behaviours around body parsing and such.